libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java

```diff
@@ -63,18 +58,19 @@ public String mediaTypeWithoutParameters() {
 
     /**
      * Parses a header value into it's parts.
-     * follows https://tools.ietf.org/html/rfc7231#section-3.1.1.1 but allows only single media type and do not support quality factors
+     * follows https://tools.ietf.org/html/rfc7231#section-3.1.1.1
+     * but allows only single media type. Media ranges will be ignored (treated as not provided)
      * Note: parsing can return null, but it will throw exceptions once https://github.com/elastic/elasticsearch/issues/63080 is done
-     * Do not rely on nulls
+     * TODO Do not rely on nulls
      *
      * @return a {@link ParsedMediaType} if the header could be parsed.
      * @throws IllegalArgumentException if the header is malformed
      */
     public static ParsedMediaType parseMediaType(String headerValue) {
-        if (DEFAULT_ACCEPT_STRING.equals(headerValue) || "*/*".equals(headerValue)) {
-            return null;
-        }
         if (headerValue != null) {
+            if (isMediaRange(headerValue) || "*/*".equals(headerValue)) {
+                return null;
+            }
```

Comment:
I think this should be moved down to and checked against `elements[0]` instead of the whole headerValue. A parameter _could_ have a comma. Probably not..but slightly more correct.

Reply:
If we check for element[0] we could not check against something like text/HTML;q=0, application/json
Media range is a coma separated list where each media type can have a list of parameters after semicolon .
Comma (delimiters at all) are not allowed within media type name or parameters

Reply:
OK ..that makes sense. sorry for the noise.

_however...minor clarification your values for the parameters may be quoted which can contain any ASCII char `  accept-extension = ";" token [ "=" ( token | quoted-string ) ]` however, this is very unlikely to encounter a comma there_

Reply:
ah.. you are totally right. I did not look into quoted string part. Indeed this can contain a comma. 
And CVS **could** be a media type that would like to have it.
we don't support that at the moment, but at some point I think it would make sense that we do.